### ISSUE TITLE
FMP: Process to monitor missing statuses and send ZSF emails

### ISSUE THREAD

**Issue by baltes** [department-of-veterans-affairs/va.gov-team#95907]
As the IVC forms team, I need to know when a submit hasn't gotten an updated PEGA status so that I can notify the user that they should submit again.

### Acceptance criteria
- Job is aware of submits without a PEGA status
- Job is aware of how long a submit has gone without a PEGA status
- Job can trigger a VA notify template to send an email when a submit has gone X days without a status
- X in the criterion above is configurable (raise alarm to Bo if this seems difficult, we could explore releasing without)

### Additional nice-to-have
It would be a bonus if we didn't have to revise this for 10-10d or other forms with supporting documents. The distinction is that those forms may have multiple uploads for one submit. Building this so that it sends a notification based on a distinct batch ID or UUID so that a user only gets one email regardless of how many files they provided, in cases where any one of those files doesn't get an updated status, might be beneficial. 

This is from a product perspective and should be validated on the engineering side, stressing again that we want to keep the implementation very lightweight.

**Comment by michaelclement** [comment]
[Here is the job](https://github.com/department-of-veterans-affairs/vets-api/blob/master/modules/ivc_champva/app/jobs/ivc_champva/missing_form_status_job.rb) that runs every 15 minutes to check Pega status

**Comment by michaelclement** [comment]
[Here is the `send_email` logic](https://github.com/department-of-veterans-affairs/vets-api/blob/dedb94b349e58b9119a804b997c6c750c9f0fb6e/modules/ivc_champva/app/controllers/ivc_champva/v1/pega_controller.rb#L55) housed in the Pega Controller

**Comment by michaelclement** [comment]
[Here is the file](https://github.com/department-of-veterans-affairs/vets-api/blob/ab8262972fe4de27829bc23f80ad07a5636a1dfd/lib/periodic_jobs.rb#L232) that controls the frequency of the status check job

**Comment by michaelclement** [comment]
Solution has been implemented. Waiting on more approvals before merging to staging (behind feature flag) to test out there.

**Comment by michaelclement** [comment]
Still waiting on platform approval. Will update as things progress.

**Comment by michaelclement** [comment]
Platform has approved and this has been merged + is on its way to staging. This functionality is behind a feature flag, so now we can test in staging and verify things work how we expect.

**Comment by michaelclement** [comment]
**Notes on enabling in production**
Once we've confirmed the behavior, we will need to add the production email template variables to S3 param store
    + Currently need one for FMP1 and 10-10d to match what we have in VANotify

For any future forms we will need to
1. Create the failure email template in VA Notify
2. Create a small PR to add it to the template map [here](https://github.com/department-of-veterans-affairs/vets-api/blob/c0007676f2cee05707f50fe94f7639a953385257/modules/ivc_champva/app/services/ivc_champva/email.rb#L7).

**Comment by michaelclement** [comment]
Following up here, I figured out the issue I was running into and was able to successfully trigger a failure email to myself over a missing Pega status:
<img width="732" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBOWU1Qnc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--99979cced417aa3ae2dd9b8b6d83da7e15f4da15/Screenshot%202024-11-13%20at%2015.23.54.png" alt="Screenshot 2024-11-13 at 15.23.54.png" />

**Comment by michaelclement** [comment]
@baltes The number of days is configurable by setting the `failure_email_threshold_days` AWS parameter (staging only, currently).

**Comment by baltes** [comment]
@michaelclement Just to confirm, what's the current value for `failure_email_threshold_days`?

**Comment by michaelclement** [comment]
@baltes 7 days currently

**Comment by michaelclement** [comment]
FMP1 and 10-10d failure templates have been pushed to VA Notify production environment. Production flipper for failure emails [is here](https://api.va.gov/flipper/features/champva_failure_email_job_enabled) (@baltes)
